app/src/main/java/org/fossasia/openevent/general/event/EventDetailsFragment.kt

```diff
@@ -454,6 +466,45 @@ class EventDetailsFragment : Fragment() {
             menuActionBar?.getItem(i)?.isVisible = !show
         }
     }
+    private fun checkForAuthentication() {
+        if (eventViewModel.isLoggedIn())
+            writefeedback()
+        else {
+            rootView.nestedContentEventScroll.longSnackbar(getString(R.string.log_in_first))
+            redirectToLogin()
+        }
+    }
+
+    private fun redirectToLogin() {
+        LoginFragmentArgs.Builder()
+            .setSnackbarMessage(getString(R.string.log_in_first))
+            .build()
+            .toBundle()
+            .also { bundle ->
+                findNavController(rootView).navigate(R.id.loginFragment, bundle, Utils.getAnimFade())
+            }
+    }
+
+    private fun writefeedback() {
+        val layout = layoutInflater.inflate(R.layout.dialog_feedback, null)
```

Comment:
Not needed directly put layout in setView of AlertDialog. I had to do this in changePassword dialog because I wanted to run code without button press.

Reply:
I think layout is being referred for getting text and add textChangedListeners. So I suppose its fine to inflate layout here